File: ruby/matrix/matrix.rb
class Matrix

  attr_accessor :rows, :columns

  def initialize(string)
    @string = string
    @rows = []
    @columns = []
    fill
  end

  private

  def fill
    @string.each_line { |l| rows << l.split.map(&:to_i) }
    rows.length.times do |i|
      columns[i] = []
      rows.each { |j| columns[i] << j[i] }
    end
  end

end

File: javascript/matrix/matrix.js
export class Matrix {
  constructor(text) {
    this.text = text;
  }

  get rows() {
    return this.text
               .split("\n")
               .map(row => row.split(' ')
                              .map(element => Number(element)));
  }

  get columns() {
    return this.rows[0]
               .map((_element, index) => this.rows.map(row => row[index])
            );
  }
}